machine/quanta/quanta_common_p2020/u-boot/platform-quanta-common_p2020.patch

```diff
@@ -0,0 +1,2235 @@
++		printf("ERROR: Changing diag. version is not permitted here.\n");
++		return(FALSE);
++		break;
+ 	case TLV_CODE_DEVICE_VERSION:
+ 		value = simple_strtoul(strval, NULL, 0);
+ 		if (value >= 256) {
+@@ -633,10 +639,8 @@ static bool tlvinfo_add_tlv(u8 * eeprom, int tcode, char * strval)
+ 		new_tlv_len = 2;
+ 		break;
+ 	case TLV_CODE_MANUF_DATE:
+-		if (set_date(data, strval) != 0) {
++		printf("ERROR: Changing manufacturing date is not permitted here.\n");
+ 			return(FALSE);
+-		}
+-		new_tlv_len = 19;
+ 		break;
+ 	case TLV_CODE_MAC_BASE:
+ 		if (set_mac(data, strval) != 0) {
+@@ -648,6 +652,9 @@ static bool tlvinfo_add_tlv(u8 * eeprom, int tcode, char * strval)
+ 		printf("WARNING: The CRC TLV is set automatically and cannot be set manually.\n");
+ 		return(FALSE);
+ 	case TLV_CODE_VENDOR_EXT:
++		strncpy(data, strval, MAX_TLV_VALUE_LEN);
++		new_tlv_len = min(MAX_TLV_VALUE_LEN, strlen(strval));
++		break;
+ 	default:
+ 		if (set_bytes(data, strval, &new_tlv_len) != 0 ) {
+ 			return(FALSE);
+@@ -749,65 +756,6 @@ static int set_mac(char *buf, const char *string)
+ }
+ 
+ /**
+- *  set_date
+- *
+- *  Validates the format of the data string
+- *
+- *  This function takes a pointer to a date string (i.e. MM/DD/YYYY hh:mm:ss)
+- *  and validates that the format is correct. If so the string is copied
+- *  to the supplied buffer.
+- */
+-static int set_date(char *buf, const char *string)
+-{
+-	int i;
+-
+-	if (!string) {
+-		printf("ERROR: NULL date string passed in.\n");
+-		return -1;
+-	}
+-
+-	if (strlen(string) != 19) {
+-		printf("ERROR: Date strlen() != 19 -- %d\n", strlen(string));
+-		printf("ERROR: Bad date format (MM/DD/YYYY hh:mm:ss): %s\n", string);
+-		return -1;
+-	}
+-
+-	for (i = 0; string[i] != 0; i++) {
+-		switch (i) {
+-		case 2:
+-		case 5:
+-			if (string[i] != '/') {
+-				printf("ERROR: Bad date format (MM/DD/YYYY hh:mm:ss): %s\n", string);
+-				return -1;
+-			}
+-			break;
+-		case 10:
+-			if (string[i] != ' ') {
+-				printf("ERROR: Bad date format (MM/DD/YYYY hh:mm:ss): %s\n", string);
+-				return -1;
+-			}
+-			break;
+-		case 13:
+-		case 16:
+-			if (string[i] != ':') {
+-				printf("ERROR: Bad date format (MM/DD/YYYY hh:mm:ss): %s\n", string);
+-				return -1;
+-			}
+-			break;
+-		default:
+-			if (!is_digit(string[i])) {
+-				printf("ERROR: Bad date format (MM/DD/YYYY hh:mm:ss): %s\n", string);
+-				return -1;
+-			}
+-			break;
+-		}
+-	}
+-
+-	strcpy(buf, string);
+-	return 0;
+-}
+-
+-/**
+  *  set_bytes
+  *
+  *  Converts a space-separated string of decimal numbers into a
+@@ -968,7 +916,7 @@ int write_sys_eeprom(void *eeprom, int len)
+  */
+ int read_sys_eeprom(void *eeprom, int offset, int len)
+ {
+-	int ret;
++	int ret, ret2 = 0;
+ #ifdef CONFIG_SYS_EEPROM_BUS_NUM
+ 	unsigned int bus;
+ #endif
+@@ -977,9 +925,29 @@ int read_sys_eeprom(void *eeprom, int offset, int len)
+ 	bus = i2c_get_bus_num();
+ 	i2c_set_bus_num(CONFIG_SYS_EEPROM_BUS_NUM);
+ #endif
++#ifdef CONFIG_MB_DETECT
```

Comment:
To avoid these global variables this function could call a new function which returns the needed i2c info.